packages/askui-nodejs/src/execution/ui-control-client.ts

```diff
@@ -721,136 +677,51 @@ export class UiControlClient extends ApiCommands {
-          }
-
-          let finalCommand = command;
-          if (element.relation) {
-            finalCommand = this.evaluateRelationExpectExistence(element, command);
-          }
-
-          returnValue.elements.push(
-            (await this.evaluateFinalExpectExistenceCommand(finalCommand, element)),
-          );
-          break;
-        }
-
-        case 'button':
-        case 'element': {
-          let command = this.get().element().withText(element.label);
-
-          if (element.type === 'button') {
-            command = this.get().button().withText(element.label);
-          }
-
-          let finalCommand = command;
-          if (element.relation) {
-            finalCommand = this.evaluateRelationExpectExistence(element, command);
-          }
-
-          returnValue.elements.push(
-            (await this.evaluateFinalExpectExistenceCommand(finalCommand, element)),
-          );
-          break;
-        }
-
-        case 'checkbox':
-        case 'switch': {
-          let command = this.get().checkbox();
-
-          switch (element.type) {
-            case 'checkbox':
-              break;
-            case 'switch':
-              command = this.get().switch();
-              break;
-            default:
-              throw new ValidationError('expected either "checkbox" or "switch"');
-          }
-
-          let finalCommand;
-          if (element.relation) {
-            finalCommand = this.evaluateRelationExpectExistence(element, command);
-          } else {
-            finalCommand = command.nearestTo().text(element.label);
-          }
-
-          returnValue.elements.push(
-            (await this.evaluateFinalExpectExistenceCommand(finalCommand, element)),
-          );
-          break;
-        }
-
-        default:
-          throw new ValidationError(
-            "No valid element type: Use one of 'text' | 'button' | 'checkbox' | 'switch' | 'textfield' | 'element'",
-          );
+    const elements = await params.reduce(async (accumulatorPromise, param) => {
+      const acc = await accumulatorPromise;
+      const command = this.get()[param.type]();
+      let finalCommand = param.text !== undefined
+        ? this.evaluateMatchingProperty(command, param.text) as FluentFiltersOrRelationsGetter
+        : command;
+      if (param.relation) {
+        finalCommand = this.evaluateRelation(
+          finalCommand,
+          param.relation.type,
+          param.relation.text,
+        ) as FluentFiltersOrRelationsGetter;
       }
-    }
-
-    returnValue.everythingExists = !(returnValue.elements.filter(
-      (el) => el.exists === false,
-    ).length > 0);
-    return new Promise<ExpectExistenceReturnValue>((resolve) => {
-      resolve(returnValue);
-    });
+      return [
+        ...acc,
+        {
+          ...param,
+          exists: (await finalCommand.exec()).length > 0,
+        },
+      ];
+    }, Promise.resolve([] as ExpectExistenceElement[]));
```

Comment:
You don't need a type cast. this is a generic (⟶ generics) method (have a look at the signature):
```suggestion
    }, Promise.resolve<ExpectExistenceElement[]>([]));
```